tests/integration/test_sharding.py

```diff
@@ -583,3 +623,329 @@ def test_asking_follower(cluster):
+
+        for i in range(len(cluster_nodes)):
+            node_data = cluster_nodes[i].split(b' ')
+
+            if local_node_id == node_data[0]:
+                assert node_data[2] == b"myself,master"
+            else:
+                assert node_data[2] == b"master"
+
+            assert node_data[3] == b"-"
+
+            if node_data[0] == stable_node_id:
+                assert node_data[8] == b"0"
+            elif node_data[0] == migrate_node_id:
+                assert node_data[8] == b"501"
+            elif node_data[0] == import_node_id:
+                assert node_data[8] == b""
+            else:
+                assert False, "failed to match id {}".format(node_data[0])
+
+    validate_nodes(cluster.node(1).execute('CLUSTER', 'NODES').splitlines())
+
+
+def test_cluster_nodes_for_multiple_slots_range_sg(cluster):
+    cluster.create(3, raft_args={'sharding': 'yes', 'external-sharding': 'yes'})
+    shardgroup_id_1 = "1" * 32
+    shardgroup_id_2 = "2" * 32
+    shardgroup_id_3 = "3" * 32
+
+    assert cluster.execute(
+        'RAFT.SHARDGROUP', 'REPLACE',
+        '3',
+
+        shardgroup_id_1,
+        '3', '1',
+        '0', '500', '1', '123',
+        '600', '700', '2', '123',
+        '800', '1000', '3', '123',
+        '%s00000001' % shardgroup_id_1, '1.1.1.1:1111',
+
+        shardgroup_id_2,
+        '2', '1',
+        '1001', '16383', '2', '123',
+        '600', '700', '3', '123',
+        '%s00000001' % shardgroup_id_2, '2.2.2.2:2222',
+
+        shardgroup_id_3,
+        '2', '1',
+        '800', '1000', '2', '123',
+        '1001', '16383', '3', '123',
+        '%s00000001' % shardgroup_id_3, '3.3.3.3:3333',
+        ) == b'OK'
+
+    def validate_nodes(cluster_nodes):
+        assert len(cluster_nodes) == 3
+        cluster_dbid = cluster.leader_node().info()["raft_dbid"]
+        node_id_1 = "{}00000001".format(shardgroup_id_1).encode()
+        node_id_2 = "{}00000001".format(shardgroup_id_2).encode()
+        node_id_3 = "{}00000001".format(shardgroup_id_3).encode()
+        local_node_id = "{}00000001".format(cluster_dbid).encode()
+
+        for i in range(len(cluster_nodes)):
+            node_data = cluster_nodes[i].split(b' ')
+
+            if local_node_id == node_data[0]:
+                assert node_data[2] == b"myself,master"
+            else:
+                assert node_data[2] == b"master"
+
+            assert node_data[3] == b"-"
+
+            if node_data[0] == node_id_1:
+                assert node_data[8] == b"0-500"
+                assert node_data[9] == b"800-1000"
+            elif node_data[0] == node_id_2:
+                assert node_data[8] == b"600-700"
+            elif node_data[0] == node_id_3:
+                assert node_data[8] == b"1001-16383"
+            else:
+                assert False, "failed to match id {}".format(node_data[0])
+
+    validate_nodes(cluster.node(1).execute('CLUSTER', 'NODES').splitlines())
```

Comment:
```suggestion
    validate_nodes(cluster.node(1).execute('CLUSTER', 'NODES').splitlines())

```
New line at the end.